apps/search/src/lib/trace-external-calls.ts

```diff
@@ -0,0 +1,52 @@
+import { ALGOLIA_TIMEOUT_MS } from "./algolia-timeouts";
+import { createLogger } from "./logger";
+
+const logger = createLogger("ExternalCall");
+
+// Slow warning thresholds for different external call types
+export const DYNAMODB_SLOW_THRESHOLD_MS = 1000;
+export const ALGOLIA_SLOW_THRESHOLD_MS = ALGOLIA_TIMEOUT_MS * 2;
```

Comment:
I dont think this can ever happen, because there is a timeout ON algolia call, so algolia lib will terminate in half a time

If we want to track this as warning, you should rather make it `ALGOLIA_TIMEOUT_MS / 2`